src/simulate.jl

```diff
@@ -7,35 +7,72 @@
 #############################################################################
 
 
-#TODO _const should not exist 
-
 using Distributions
 
 type NoiseLaw
-    supportSize::Int64 
+    # Number of points in distribution:
+    supportSize::Int64
+    # Position of points:
     support::Array{Float64,2}
+    # Probabilities of points:
     proba::Vector{Float64}
 end
 
-function NoiseLaw_const(supportSize,support,proba)
+
+"""
+Instantiate an element of NoiseLaw
+
+
+Parameters:
+- supportSize (Int64)
+    Number of points in discrete distribution
+
+- support
+    Position of each point
+
+- proba
+    Probabilities of each point
+
+
+Return:
+- NoiseLaw
+
+"""
+function NoiseLaw_const(supportSize, support, proba)
     supportSize = convert(Int64,supportSize)
     if ndims(support)==1
         support = reshape(support,1,length(support))
     end
-    
+
     if ndims(proba) == 2
         proba = vec(proba)
     elseif  ndims(proba) >= 2
         proba = squeeze(proba,1)
-    end    
-    
-    if sum(proba) !=1 error("probability doesnot sum to 1") end
```

Comment:
Perhaps let a line of code like this
if abs( sum(proba)-1 ) != 0 println("Warning, the probability sums to",  sum(proba))